### PR TITLE
Change XMSS License from `(Apache 2.0 AND MIT)` to `(Apache 2.0 OR MIT) AND CC0-1.0`

### PR DESCRIPTION
**Pull request by ducnguyen-sb**
According to liboqs maintainer request, this PR change the license of XMSS contribution from 
`(Apache 2.0 AND MIT)` to `(Apache 2.0 OR MIT) AND CC0-1.0`.

1. Moved standalone LICENSE-MIT into LICENSE file.
2. A few newlines, spaces format. 

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: src/sig_stfl/xmss/LICENSE [hunk 1]
```diff
@@ -1,8 +1,18 @@
 ## License
 
-This XMSS reference implementation is Copyright (c) 2024 SandboxAQ and licensed under both the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0.txt) and [MIT License](LICENSE-MIT).
+This XMSS reference implementation is Copyright (c) 2024 SandboxAQ and licensed under the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0.txt) OR MIT License at your option.
```
Comment (baentsch) [inline src/sig_stfl/xmss/LICENSE:3]: Mirroring the xmss.md file, shouldn't there also be reference to the original authors (and the(ir) CC0 license) in this file?

Comment (ducnguyen-sb) [reply src/sig_stfl/xmss/LICENSE:3]: Oh, yes, you're right. Let me add that.

Comment (ducnguyen-sb) [reply src/sig_stfl/xmss/LICENSE]: updated in https://github.com/open-quantum-safe/liboqs/pull/1697/commits/983e6acd65f71dd35963c0f0ac00b0f9bcff89c2

**Review by baentsch — COMMENTED** [review]
Thanks for this update! I accept the blame for causing this work but I honestly think this is the right way to integrate the work (and honor the intentions) of the original authors. LGTM except one nit (see separate comment).

**Review by baentsch — APPROVED** [review]

**Comment by ducnguyen-sb** [comment]
@baentsch , no worry, thank you very much for reviewing the PR. Without you I couldn't catch these problems myself.

**Pull request merged**